File: CodingSample/PreReading/ReflectionTest/FinLib/Finance/HomeLoan.cs
namespace Finance;

public class HomeLoan
{
	public double Common(double amount, int period) => amount < 5000000 ? 9 : 8.5;

	public double Woman(double amount, int period) => Common(amount, period) - 1;
	
	public double Welfare(double amount, int period) => 0.5 * Common(amount, period);
}

File: CodingSample/Java/Core/Runtime/Reflection/bankfin/src/main/java/finance/HomeLoan.java
package finance;

public class HomeLoan {

    public float common(double amount, int period) {
        return amount < 5000000 ? 9 : 8.5f;
    }

    public float woman(double amount, int period) {
        return common(amount, period) - 1;
    }

    @MaxDuration(12)
    public float welfare(double amount, int period) {
        return common(amount, period) / 2;
    }
}
